lib/cli.mjs

```diff
@@ -368,10 +369,22 @@ export default async function main(argv) {
 
 	let formatter = undefined;
 
+	const cwd = process.cwd();
+	const resolvedConfig = await resolveConfig(cwd, { configFile, configBasedir });
+	const configFormatter = resolvedConfig?.formatter;
+	const cliFlagExists = Object.prototype.hasOwnProperty.call(cli.flags, 'formatter');
+	const configFormatterExists = configFormatter && formatters[configFormatter];
+
 	if (isString(customFormatter)) {
 		formatter = await importCustomFormatter(customFormatter);
-	} else if (isString(formatterInput)) {
+	} else if (cliFlagExists && isString(formatterInput)) {
 		formatter = /** @type {import('stylelint').FormatterType} */ (formatterInput);
+	} else if (configFormatter && !configFormatterExists) {
+		formatter = await importCustomFormatter(configFormatter);
+	} else if (configFormatter) {
+		formatter = /** @type {import('stylelint').FormatterType} */ (configFormatter);
+	} else {
+		formatter = DEFAULT_FORMATTER;
 	}
```

Comment:
[suggest] I now understand `cli.mjs` must load a configuration to determine the default formatter. This would make the code more complex than necessary.

So, how about adding an internal default formatter option to `LinterOptions`? This could allow the default formatter logic to be determined in one place in `standalone.mjs`. For example:

```diff
export type LinterOptions = {
		// ...
 		formatter?: FormatterType | Formatter;
+		/** @internal */
+		_defaultFormatter?: FormatterType;
		// ...
};
```

Reply:
question

I don't know about @ybiquitous solution but I agree that something is wrong about this.
i.e. it seems to me that there's code that doesn't belong in cli.mjs or
Maybe an util could be created that would return the formatter?
e.g. `getFormatter(cli.flags)`

Reply:
I opted for @Mouvedia approach.